doc/productivity.md

```diff
@@ -51,6 +64,12 @@ If you have multiple threads on your machine, you can tell `make` to utilize all
 make -j"$(($(nproc)+1))"
 ```
 
+### Only build what you need
+
+When rebuilding incrementally while developing, be aware that running `make` with no targets will do a _vast_ amount of work, most of which you probably don't care about. In particular, it will build the GUI (unless you've disabled it) and all the tests (which take a significant multiple of the time to build the application itself.)
+
+Obviously it is important to build and run the tests at appropriate times -- but when you just want a quick compile to check your work, consider `make src/bitcoind` or similar.
```

Comment:
Could mention that can make multiple targets and/or give more examples:
 - `make -j src/bitcoind src/bitcoin-cli`
 - `make -j src/qt/bitcoin-qt`
 - `make -j -C src bitcoin_bench`

Reply:
Done!

Reply:
Nice.